From f400c4c1e2589eae5678ff1134cbd3bcd77d468d Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Wed, 22 Apr 2015 16:12:39 -0700
Subject: [PATCH] Add MMC_SDHCI_ACPI_FORCE_POWER_ON config option

Adding the MMC_SDHCI_ACPI_FORCE_POWER_ON config option that was missed
in the eMMC driver forklift.
---
 drivers/mmc/host/Kconfig |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index 7fc5099..bdb89e1 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -93,6 +93,21 @@ config MMC_SDHCI_ACPI
 
 	  If unsure, say N.
 
+config MMC_SDHCI_ACPI_FORCE_POWER_ON
+	bool "Force Power On through ACPI methods"
+	depends on MMC_SDHCI_ACPI
+	help
+	  This forces the controller and all connected devices to
+	  power on before probing. This is done using ACPI.
+
+	  Some devices may have separate enabling lines that are
+	  controlled through GPIO. By calling the ACPI _PS0 method,
+	  these devices will be powered on and enabled before
+	  probing. This is to ensure all devices connected can be
+	  seen by the controller.
+
+	  If unsure, say N.
+
 config MMC_SDHCI_PLTFM
 	tristate "SDHCI platform and OF driver helper"
 	depends on MMC_SDHCI
-- 
1.7.9.5

